### PR TITLE
Refactored methods which dealed directly with gridfile contents instead ...

### PR DESCRIPTION
**Pull request by jsbattig**
...of using gridfile object. This refactoring is needed in order to later refactore pre and post processing of chunks logic to optimize buffer allocation and prepare the ground for effective usage of pre and post processing for encryption, which requires keeping a context between calls

### CODE REVIEW

File: src/gridfs.h [hunk 1]
```diff
@@ -114,7 +131,7 @@ MONGO_EXPORT void gridfile_writer_init( gridfile *gfile, gridfs *gfs, const char
  *  stream to a GridFS file. When finished, be sure to call gridfs_writer_done.
  *
```
Comment (paulmelnikow) [inline src/gridfs.h]: Would you add a comment explaining what the return value means?

**Comment by paulmelnikow** [comment]
Looks good to me!

File: src/gridfs.h (hunk 1 above)
Comment (jsbattig) [inline src/gridfs.h]: Good catch!

I will...

**Comment by gjmurakami-10gen** [comment]
This pull request cannot be automatically merged.  I must have pull in something with a conflict.  You can resync and resubmit if I don't get around to a manual merge in a timely fashion.

**Comment by jsbattig** [comment]
I will try to make some time to re-sync and re-submit.

## 

José Sebastián Battig
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by paulmelnikow** [comment]
You could also try what worked for me: merge master into your branch and resolve the conflicts, then push the merge commit to your branch.

**Comment by jsbattig** [comment]
Done. I'll commit.

## 

José Sebastián Battig
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by jsbattig** [comment]
Done

## 

José Sebastián Battig
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Pull request merged**